build.gradle.kts

```diff
@@ -17,7 +17,7 @@
 
 plugins {
 	id("java")
-	id("org.springframework.boot") version "2.4.0"
+	id("org.springframework.boot") version "2.3.9.RELEASE"
```

Comment:
@sbilello you can use the `org.springframework.boot` Gradle Plugin 2.4.+, latest is 2.4.5, with Spring Boot 2.3.+. The version of the plugin is independent from the version of the framework.

Reply:
![SpringBoot2 4 0](https://user-images.githubusercontent.com/1815376/117337401-2cf90280-ae52-11eb-80f6-82855abee9b7.png)
![SpringBoot2 3 9 RELEASE](https://user-images.githubusercontent.com/1815376/117337406-2e2a2f80-ae52-11eb-826d-5732207c42cd.png)
@berngp I mentioned that upgrading the plugin to `spring 2.4.+` fixes the issue. 
This is not a blocker for me and I opened this issue to help someone googling the problem and finding the corresponding solution :) . 
I just reported because I remember a conversation where it was required to keep compatibility with spring 2.3.X
However, the plugin affects the dependencies in the classPath and I see the match between the spring-boot plugin version and the spring-boot-starter-web version.
Cheers!

Reply:
Thanks @sbilello, I stand corrected. I see that now via Gradle's dependencyInsights...

## Command
`./gradlew  -q dI  --dependency org.springframework.boot:spring-boot-starter-web`

## Output:

```
org.springframework.boot:spring-boot-starter-web:2.4.5
   variant "apiElements" [
      org.gradle.category            = library
      org.gradle.dependency.bundling = external
      org.gradle.jvm.version         = 8
      org.gradle.libraryelements     = jar (compatible with: classes)
      org.gradle.usage               = java-api
      org.gradle.status              = release (not requested)
   ]
   Selection reasons:
      - Selected by rule
      - By constraint

org.springframework.boot:spring-boot-starter-web -> 2.4.5
\--- compileClasspath

org.springframework.boot:spring-boot-starter-web:{prefer 2.3.9.RELEASE} -> 2.4.5
\--- com.netflix.graphql.dgs:graphql-dgs-platform-dependencies:3.12.1
     \--- compileClasspath (requested com.netflix.graphql.dgs:graphql-dgs-platform-dependencies:latest.release)
```

Reply:
@sbilello the plugin will downgrade the version of Kotlin, go figure.
To make this work you will need to explicitly bring Kotlin 1.4.+, e.g. below.

```
implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.4.31")
implementation("org.jetbrains.kotlin:kotlin-stdlib-common:1.4.31")
implementation("org.jetbrains.kotlin:kotlin-stdlib:1.4.31")
implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.4.31")
```

I think we can address this with the BOM. Will send a PR on the DGS Framework and CC you.
Thanks for catching this!

Reply:
I didn't have the time to dig into it... but it was for sure some transitive dependency problem! Thanks for fixing it!

Reply:
As an option you could  specify `extra["kotlin.version"] = "1.4.31"` and not have to list the kotlin libraries explicitly.
This is documented in the Spring Boot Gradle Plugin page, [here](https://docs.spring.io/spring-boot/docs/2.4.4/gradle-plugin/reference/htmlsingle/#managing-dependencies-dependency-management-plugin-customizing). Still boggles the mind that it downgrades the version of a transitive that is explicitly set.

Reply:
It is also really unfortunate that even if we, the DGS framework, publish the proper metadata, ref below, the Spring Boot Gradle Plugin will force an older version.

Gradle metadata
```
{
          "group": "org.jetbrains.kotlin",
          "module": "kotlin-stdlib-jdk8",
          "version": {
            "requires": "1.4.32"
          }
}
```

Maven POM
```
<dependency>
      <groupId>org.jetbrains.kotlin</groupId>
      <artifactId>kotlin-stdlib-jdk8</artifactId>
      <version>1.4.32</version>
      <scope>compile</scope>
</dependency>
```

I'm on the fence, thinking on removing the plugin from the example(s).